schematools/contrib/django/factories.py

```diff
@@ -27,6 +27,212 @@
 TypeAndSignature = Tuple[Type[models.Field], tuple, Dict[str, Any]]
 
 
+class RelationMaker:
+    """ Superclass to generate info for relation fields """
+
+    def __init__(
+        self,
+        dataset: DatasetSchema,
+        table: DatasetTableSchema,
+        field: DatasetFieldSchema,
+        field_cls,
+        *args,
+        **kwargs,
+    ):
+        self.dataset = dataset
+        self.table = table
+        self.field = field
+        self._field_cls = field_cls
+        self._args = args
+        self._kwargs = kwargs
+        self.relation = field.relation or field.nm_relation
+        self.fk_relation = field.relation
+        self.nm_relation = field.nm_relation
+
+    @classmethod
+    def fetch_relation_parts(cls, relation):
+        relation_parts = relation.split(":")
+        return relation_parts[:2]
+
+    @classmethod
+    def is_loose_relation(cls, relation, dataset, field):
+        """Determine if relation is loose or not."""
+
+        related_dataset_id, related_table_id = cls.fetch_relation_parts(relation)
+        related_dataset = dataset.get_dataset_schema(related_dataset_id)
+        related_table = related_dataset.get_table_by_id(related_table_id)
+
+        # Short-circuit for non-temporal or on-the-fly (through or nested) schemas
+        if (
+            not related_table.is_temporal
+            or field._parent_table.is_through_table
+            or field._parent_table.is_nested_table
+        ):
+            return False
+
+        # So, target-side of relation is temporal
+        # Determine fieldnames used for temporal
+        sequence_identifier = related_table.temporal["identifier"]
+        identifier = related_dataset.identifier
+
+        # If temporal, this implicates that the type is not a scalar
+        # but needs to be more complex (object) or array_of_objects
+        if field.type in set(["string", "integer"]) or field.is_array_of_scalars:
```

Comment:
`set` seems superfluous here.

Reply:
Yeah, some premature optimization for a two item list. Building the set is probably more expensive than scanning the list. I'll remove it.